### PR TITLE
tests: update lxd-state tool to install and prepare the lxd snap

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The idea is to simplify the tests by using a tool which installs and configures lxd.

### CODE REVIEW

File: tests/lib/tools/lxd-state [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+    echo "usage: lxd-state undo-mount-changes"
+    echo "       lxd-state prepare-snap"
+}
+
+prepare_snap(){    
+    echo "lxd-state: installing lxd snap"
+    snap install lxd --channel="$LXD_SNAP_CHANNEL"
+    if tests.pkgs is-installed lxd; then
+        echo "lxd-state: remove any installed pkgs (some images carry them) to ensure we use the snap"
+        tests.pkgs remove lxd lxd-client
```
Comment (Meulengracht) [inline tests/lib/tools/lxd-state:13]: This fails on distributions that do not have the lxd-client package

```
lxd-state: remove any installed pkgs (some images carry them) to ensure we use the snap
+ apt-get remove --yes lxd lxd-client
Reading package lists...
Building dependency tree...
Reading state information...
Package 'lxd' is not installed, so not removed
E: Unable to locate package lxd-client
```

Comment (sergiocazzolato) [reply tests/lib/tools/lxd-state]: Nice, thanks for the review, the issue should be fixed now.

**Review by Meulengracht — APPROVED** [review]
LGTM

**Review by mvo5 — APPROVED** [review]
Very nice, thank you!

**Pull request merged**